### PR TITLE
Specification does not define variablesReference lifetime

### LINKED ISSUE #35: Specification does not define variablesReference lifetime
The specification does not define when `variablesReference` numbers become invalid and when previously used numbers can be used again without adverse or unexpected consequences.

### PR DESCRIPTION
**Pull request by weinand**
Fixes #35

### CODE REVIEW

**Review by connor4312 — APPROVED** [review]
One suggested wording tweak (so disabling auto merge) but lgtm overall

File: overview.md [hunk 1]
```diff
@@ -184,6 +184,12 @@ Threads
 
 The value of variables can be modified through the [**setVariable**](./specification#Requests_SetVariable) request.
 
+#### Lifetime of Objects References
+
+Complex structural objects such as [`scopes`](./specification#Types_Scope) or [`variables`](./specification#Types_Variable) are not embedded directly in their containers ([`stack frames`](./specification#Types_StackFrame), [`scopes`](./specification#Types_Scope), [`variables`](./specification#Types_Variable)), but must be retrieved with separate [**scopes**](./specification#Requests_Scopes) and [**variables**](./specification#Requests_Variables) requests based on *object references*. An object reference is an integer in the open interval (0, 2<sup>31</sup>) assigned to objects by the debug adapter.
```
[the remark this answers is not in the document]
Comment (weinand) [reply overview.md:189]: excellent point, thanks.

[the remark this answers is not in the document]
Comment (weinand) [reply overview.md]: Thanks! I've added a sentence to address your concern.

**Review by puremourning — APPROVED** [review]
LGTM

**Review by connor4312 — APPROVED** [review]

**Pull request merged**